doc/doxygen/public/coding-conventions.doxygen

```diff
@@ -0,0 +1,573 @@
+/**
+
+  @page coding_conventions Coding Conventions
+
+  Use the following code conventions when contributing to OpenMS.
+
+  OpenMS uses coding conventions that are automatically checked using <code>cpplint</code> (<code>/src/tests/coding/cpplint.py</code>). You can find a configuration file for the <code>CLion IDE</code> here. You can import it by selecting <B>Preferences</B> > <B>Code Style</B> > <B>Manage</B>.
+
+  @tableofcontents
+
+  @section formatting_and_style Formatting and style
+
+  The following section focuses on formatting and style.
+
+  @subsection indentation Indentation
+
+  Use two spaces to indent. Tabulators in string literals should use the escape character <code>\t</code>.
+
+  @subsection spaces Spaces
+
+  Use spaces after built-in key words (e.g. <code>for</code>, <code>if</code>, <code>else</code>, etc.), and before and after binary mathematical operators.
+
+  @subsection line_endings Line endings
+
+  Native line endings should be allowed on each platform. This is desirable as Visual Studio for example will always insert <code>CRLF</code> even if the file is <code>LF</code> only, leading to a mixed line ending style for this file. Native <code>eol</code> style avoids this problem.
+
+  @subsection bracket_placements Bracket placements
+
+  Matching pairs of opening and closing curly braces should be set to the same column. See the following example:
+
+  @code{cpp}
+    while (continue == true)
```

Comment:
let's use `@code{.cpp}`, and indent the code to left. does this work? `{cpp}`. https://www.doxygen.nl/manual/commands.html#cmdcode

<img width="491" alt="Screenshot 2022-06-03 at 10 56 53 PM" src="https://user-images.githubusercontent.com/4143778/171915417-8408e0a9-853c-4643-8936-fe6012a88733.png">

Reply:
let's use `.cpp` for consistency.